contracts/Verifiers/NftLimitSwapVerifier.sol

```diff
@@ -0,0 +1,112 @@
+  /// @param bitmapIndex The index of the replay bit's bytes32 slot [signed]
+  /// @param bit The value of the replay bit [signed]
+  /// @param tokenIn The input token provided for the swap. Can be ERC20 or Native [signed]
+  /// @param nftOut The ERC721 output token required to be received from the swap [signed]
+  /// @param tokenInAmount Amount of tokenIn provided [signed]
+  /// @param nftOutAmount Amount of nftOut required to be received [signed]
+  /// @param expiryBlock The block when the swap expires [signed]
+  /// @param to Address of the contract that will fulfill the swap [unsigned]
+  /// @param data Data to execute on the `to` contract to fulfill the swap [unsigned]
+  function tokenToNft(
+    uint256 bitmapIndex, uint256 bit, address tokenIn, IERC721 nftOut, uint256 tokenInAmount, uint256 nftOutAmount,
+    uint256 expiryBlock, address to, bytes calldata data
+  )
+    external
+  {
+    require(expiryBlock > block.number, 'Expired');
+  
+    Bit.useBit(bitmapIndex, bit);
+
+    uint256 nftOutBalance = nftOut.balanceOf(address(this));
+
+    if (tokenIn.isEth()) {
+      require(tokenInAmount <= address(this).balance, 'InsufficientEth');
+      CALL_EXECUTOR.proxyCall{value: tokenInAmount}(to, data);
+    } else {
+      IERC20(tokenIn).transfer(to, tokenInAmount);
+      CALL_EXECUTOR.proxyCall(to, data);
+    }
+
+    uint256 nftOutAmountReceived = nftOut.balanceOf(address(this)) - nftOutBalance;
+    require(nftOutAmountReceived >= nftOutAmount, 'NotEnoughReceived');
+  }
+
+  /// @dev Executes swap from a single ERC721 ID to fungible token (ERC20 or Native)
+  /// @notice This should be executed by metaDelegateCall() or metaDelegateCall_EIP1271() with the following signed and unsigned params
+  /// @param bitmapIndex The index of the replay bit's bytes32 slot [signed]
+  /// @param bit The value of the replay bit [signed]
+  /// @param nftIn The ERC721 input token provided for the swap [signed]
+  /// @param tokenOut The output token required to be received from the swap. Can be ERC20 or Native [signed]
+  /// @param nftInID The ID of the nftIn token provided [signed]
+  /// @param tokenOutAmount Amount of tokenOut required to be received [signed]
+  /// @param expiryBlock The block when the swap expires [signed]
+  /// @param to Address of the contract that will fulfill the swap [unsigned]
+  /// @param data Data to execute on the `to` contract to fulfill the swap [unsigned]
+  function nftToToken(
+    uint256 bitmapIndex, uint256 bit, IERC721 nftIn, address tokenOut, uint256 nftInID, uint256 tokenOutAmount,
+    uint256 expiryBlock, address to, bytes calldata data
+  )
+    external
+  {
+    require(expiryBlock > block.number, 'Expired');
+  
+    Bit.useBit(bitmapIndex, bit);
+
+    uint256 tokenOutBalance = tokenOut.balanceOf(address(this));
+
+    nftIn.transferFrom(address(this), to, nftInID);
```

Comment:
I think because the `transferFrom` call is a `CALL`, there's not much that a badly written hook in the ERC721 transfer fns could do. I don't think there's any possibility of arbitrary code execution on the proxy account. Same with ERC20 transfer in LimitSwapVerifier